src/NUnitFramework/framework/Attributes/RangeAttribute.cs

```diff
@@ -47,6 +47,9 @@ public class RangeAttribute : ValuesAttribute
         /// <param name="step"></param>
         public RangeAttribute(int from, int to, int step)
         {
+            Guard.ArgumentValid(step > 0 && to >= from || step < 0 && to <= from,
```

Comment:
Me bad, sorry, I am so used to if(...) throw that I inverted the guard in my head. This is right, I am merging.